apisix/cli/ops.lua

```diff
@@ -238,9 +238,19 @@ Please modify "admin_key" in conf/config.yaml .
         util.die("missing apisix.proxy_cache for plugin proxy-cache\n")
     end
 
-    --support multiple ports listen, compatible with the original style
+    -- support multiple ports listen, compatible with the original style
     if type(yaml_conf.apisix.node_listen) == "number" then
-        local node_listen = {yaml_conf.apisix.node_listen}
+        local node_listen = {{port = yaml_conf.apisix.node_listen}}
+        yaml_conf.apisix.node_listen = node_listen
+    elseif type(yaml_conf.apisix.node_listen) == "table" then
+        local node_listen = {}
+        for index, value in ipairs(yaml_conf.apisix.node_listen) do
+            if type(value) == "number" then
```

Comment:
Is the number type really needed? I think the following way is enough.

```yaml
node_listen:
  - port: 9080
  - port: 9180
    enable_http2: true
```

Reply:
We need to be backward compatibility.

Reply:
I mean the element type inside `node_listen`, not the `node_listen` itself.

Reply:
We can also be compatible with this format:

```yaml
node_listen:
    - 9080
    - 9081
```